### PR TITLE
Adds test to verify Hypershift Install

### PR DESCRIPTION
**Pull request by MrSantamaria**

### CODE REVIEW

File: pkg/e2e/openshift/hypershift/verifyInstall.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+func init() {
+	alert.RegisterGinkgoAlert(hypershiftInstallVerify, "SD-CICD", "Diego Santamaria", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+// Checks the installation of the hypershift cluster
+var _ = ginkgo.Describe(hypershiftInstallVerify, func() {
+	ginkgo.Context("Verify install using oc", func() {
+		util.GinkgoIt("Worker nodes are ready", func(ctx context.Context) {
```
Comment (jbpratt) [inline pkg/e2e/openshift/hypershift/verifyInstall.go:33]: This test doesn't seem to be doing much, can we combine these two into one and use `checkWorkerNodesInCluster` as a requirement for checking nodes in aws?

Right now, if this first test fails to get the nodes for whatever reason, the next test will still try to run and fail so having a dependency between the two isn't the best.

I was thinking something like this:

```
Describe("hypershift", func() {
  It("worker nodes are available in aws", func() {
    By("getting the worker nodes")
    nodes, err := checkWorkerNodesInCluster(ctx)
    // expect no err
    Expect(nodes).ToNot(BeEmpty())

    By("listing the nodes in aws")
    // checkWorkerNodesInAWS()
    
    By("asserting they are equal")
    // assert the lists match
  })
})
```

**Review by jbpratt — CHANGES_REQUESTED** [review]

File: pkg/common/aws/ec2.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+// Hypershift Test Helper Function:
+// This function is used to validate the worker nodes displayed by the cluster are the same as the worker nodes displayed by the AWS account.
+func (CcsAwsSession *ccsAwsSession) CheckIfEC2ExistBasedOnNodeName(nodeName string) (bool, error) {
+	var err error
+	CcsAwsSession.GetAWSSessions()
```
[the remark this answers is not in the document]
Comment (MrSantamaria) [reply pkg/common/aws/ec2.go:12]: Upcoming in next commit. Good catch, I looked at comment dumbfounded for a good amount of time.

File: pkg/e2e/openshift/hypershift/verifyInstall.go (hunk 1 above)
Comment (MrSantamaria) [reply pkg/e2e/openshift/hypershift/verifyInstall.go:33]: Applied in upcoming commit. 
Using the By is much better than how I had set it up.

File: pkg/e2e/openshift/hypershift/verifyInstall.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+	ginkgo.Context("Verify install using AWS", func() {
+		util.GinkgoIt("Worker nodes are present in CCS account", func(ctx context.Context) {
+			Expect(checkWorkerNodesInAWS(ctx)).To(BeTrue())
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
+	})
+})
+
+// checkWorkerNodesInCluster returns a list of nodes in the cluster
+func checkWorkerNodesInCluster(ctx context.Context) ([]string, error) {
```
[the remark this answers is not in the document]
Comment (MrSantamaria) [reply pkg/e2e/openshift/hypershift/verifyInstall.go:46]: Updated in next commit.

File: pkg/e2e/openshift/hypershift/verifyInstall.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+	// add the worker nodes to the list
+	for _, node := range nodes.Items {
+		workerNodes = append(workerNodes, node.Name)
+	}
+
+	return workerNodes, nil
+}
+
+func checkWorkerNodesInAWS(ctx context.Context) (bool, error) {
```
[the remark this answers is not in the document]
Comment (MrSantamaria) [reply pkg/e2e/openshift/hypershift/verifyInstall.go:72]: Copy/paste sins

**Review by jbpratt — CHANGES_REQUESTED** [review]
two small things and LGTM!

thanks for the updates

File: pkg/e2e/openshift/hypershift/verifyInstall.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	workerNodes             []string
+)
+
+func init() {
+	alert.RegisterGinkgoAlert(hypershiftInstallVerify, "SD-CICD", "Diego Santamaria", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+// Checks the installation of the hypershift worker nodes in CCS AWS account
+var _ = ginkgo.Describe(hypershiftInstallVerify, func() {
+	defer ginkgo.GinkgoRecover()
```
[the remark this answers is not in the document]
Comment (MrSantamaria) [reply pkg/e2e/openshift/hypershift/verifyInstall.go]: It is not but it passed on a random run. 
This actually comes from a bug we introduced passing the ctx go the goroutine without a timeout. https://issues.redhat.com/browse/SDCICD-841

File: pkg/e2e/openshift/hypershift/verifyInstall.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/aws"
+	"github.com/openshift/osde2e/pkg/common/cluster"
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/util"
+)
+
+var (
+	hypershiftInstallVerify = "[Suite: hypershift]"
+	workerNodes             []string
```
[the remark this answers is not in the document]
Comment (MrSantamaria) [reply pkg/e2e/openshift/hypershift/verifyInstall.go]: Done in commit.

**Pull request merged**